AttributedLabel+Link.swift

```diff
@@ -0,0 +1,49 @@
+import Foundation
+
+extension AttributedLabel {
+    public enum LinkDetectionType: Equatable, Hashable {
+        case date
+        case address
+        case link
+        case phoneNumber
+
+        var checkingType: NSTextCheckingResult.CheckingType {
+            switch self {
+            case .date: return .date
+            case .address: return .address
+            case .link: return .link
+            case .phoneNumber: return .phoneNumber
+            }
+        }
+    }
+
+    public struct Link: Equatable, Hashable {
+        internal var perform: () -> Void
+
+        public private(set) var hashValue: Int
+        public var range: NSRange
+
+        public init(url: URL, range: NSRange) {
+            hashValue = url.hashValue
+            self.range = range
+
+            perform = {
+                UIApplication.shared.open(url, options: [:], completionHandler: nil)
```

Comment:
Thought: It might be nice to read a `URLHandler` from the `Environment`, which can default to `UIApplication`. That allows for easy stubbing of the handler (eg, maybe using it to show an in-app browser, etc).

Reply:
Good idea

Reply:
thats why they pay me the medium bucks